recipes/gtsam/all/patches/use-conan-cmake.patch

```diff
@@ -0,0 +1,12 @@
+--- CMakeLists.txt	2019-10-07 20:05:09.000000000 -0700
++++ CMakeLists_fix.txt	2020-01-15 17:38:59.097017080 -0800
+@@ -1,6 +1,9 @@
+ project(GTSAM CXX C)
+ cmake_minimum_required(VERSION 3.0)
+ 
++include(../conanbuildinfo.cmake)
```

Comment:
prefer using cmake wrapper instead of a patch. For instance: https://github.com/conan-io/conan-center-index/blob/70ed80e49ca7d5c402b1fb3f82c4675aeffb9906/recipes/expat/all/CMakeLists.txt

Reply:
If I use the wrapper (you can see that the file is there since the first commit) I get a lot of errors during compilation regarding finding header files. I'm not really sure how to fix that and that's why I decided to use this approach.

Reply:
do you have log with these errors?

Reply:
I don't but I left the CMake wrapper there so its easily to reproduce. Just comment the line where the patch is applied and replace
`self._cmake.configure(source_folder=self._source_subfolder, build_folder=self._build_subfolder)`
with 
`self._cmake.configure(build_folder=self._build_subfolder)`

Reply:
I think gtsam fails to build as a subproject because it uses `CMAKE_SOURCE_DIR` instead of `PROJECT_SOURCE_DIR`.

Reply:
@madebr it makes sense!

Reply:
Yes I though about that but I think is cleaner and simpler to just modify the main CMakeList to call  conan functions. Otherwise we can try to replace this with a patch
`gtsam/CMakeLists.txt:148:  $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>
gtsam/CMakeLists.txt:153:  $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/CppUnitLite>
gtsam/CMakeLists.txt:157:    $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/gtsam/3rdparty/metis/include>
gtsam/CMakeLists.txt:158:    $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/gtsam/3rdparty/metis/libmetis>
gtsam/CMakeLists.txt:159:    $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/gtsam/3rdparty/metis/GKlib>
`
With PROJECT_SOURCE_DIR